### PR TITLE
Allow SLIDEs to contain more than one path with varying durations

### PR DESCRIPTION
**Pull request by LumpBloom7**
This allows more than one SLIDE to originate from the same SLIDETAP. 

As another benefit, it probably makes implementing the editor a bit more friendly.

Slides with multiple patterns won't show in converts.

### CODE REVIEW

**Comment by LumpBloom7** [comment]
Note to self
https://github.com/LumpBloom7/sentakki/blob/42e92e319e6b8076f24fe532cae0346aca9e001c/osu.Game.Rulesets.Sentakki/Objects/SentakkiSlidePath.cs#L13

This is prone to race conditions when first entering song select.
`BeatmapInfoWedge` and `DifficultyCalculator` could both potentially access this value at the same time. Only a problem if the `SliderPath` cache is invalid, in which it'll malform the cache and return garbage distance values, until a restart is performed and the collision doesn't happen again.

A simple lock when accessing `VALIDPATHS` should suffice since this problem only occurs during the first time accessing that paticular path during conversion, and the cache is likely to remain valid afterwards since the control point array never changes.

In short: `SliderPath` is not thread safe.
